### DISCUSSION TITLE
How can we not force the default locale inside every URL in Next.13?

### DISCUSSION THREAD

**Discussion by MohammadMiras** [vercel/next.js#45366] [Help]
Let's say I want to create a bilingual website in English and French.

Based on the [current docs](https://beta.nextjs.org/docs/guides/internationalization), I should create a `[lang]` directly beneath `/app` directory.

This means that my URLs would have the prefix, even for the default locale.

Let's say my default locale is English.

I want my URLs to be `example.com` and `example.com/blog` for English, and `example.com/fr` and `example.com/fr/blog`.

But in this case, my URLs for the English website would be `example.com/en` and `example.com/en/blog`, which is not just ugly, but puts a lot of burden on our current SEO. Our main URLs are already indexed and we can't afford to redirect all of them to new URLs for our default locale.

What can we do for that?

**Comment by nabildroid** [comment]
use middleware to rewrite URLs

**Reply by silas-v1** [reply]
@nabildroid that middleware is not rewriting but redirecting; which is a ridiculous cost to add.

**Comment by guitheengineer** [comment]
Set the default locale on your next.config. This works because **the default locale does not have a prefix.**

For example this:
```js
// next.config.js
module.exports = {
  i18n: {
    locales: ['en-US', 'fr', 'nl-NL'],
    defaultLocale: 'en-US',
  },
}
```

For a blog page will result in: 

    /blog <- en-US
    /fr/blog <- fr
    /nl-nl/blog <- nl-NL

**Comment by AwaisTariq18** [comment]
@MohammadMiras  do we have the solution for this problem yet?

**Comment by MohammadMiras** [comment]
@guilhermefront thank you for the guidance. But that is for Next.js 12. I asked about Next.js 13.

**Comment by MohammadMiras** [comment]
@AwaisTariq18 not yet.
